### DISCUSSION TITLE
Cleaning locals.user

### DISCUSSION THREAD

**Discussion by rushanepam** [Sirneij/rust-auth#19] [Q&A] [answered]
Hi Sirnejj,

Sorry for a stupid  question, but still: when user logs out - where do we clean up the `locals.user` obj, eg `locals.user = null`. Don't we need to do so? 

And another question would be: if handle fn inside hooks.server.ts gets called every request to a sveltekit's server, does it mean that we will request for user details (`event.fetch('${BASE_API_URI}/users/current-user/')`) every time?This is smth confusing me in this flow.

Thank you in advance

**Comment by Sirneij** [comment, marked as answer]
Hi rushanepam,

In the logout function, we are not setting `locals.user = null` because it's not required. What was required was clearing the cookie since that was what validated the requests sent to the server each time a request was made. However, you can "clean" it up if you want but currently, when a user logs out and the cookie is "eaten", `local.user` automatically becomes empty.

The handle function is called every time a request is made to sveltekit's server. However, as long as the `local.user` still has some data, it won't fetch the user's details from the backend server. This is what these [lines](https://github.com/Sirneij/rust-auth/blob/320f829b8c58d682f48536e06bd9a3d9095795af/frontend/src/hooks.server.ts#L6C2-L9C3) do.

**Reply by rushanepam** [reply]
Hi Sirnejj, thank you so much for a quick response! 

However i do not understand how does `local.user` gets automatically emptied when you clean a session cookie... Could you please suggest where to look at or what to read on this matter? I am srtill new to a sveltekit and there is not too much info in official docs on the way it works... 

So maybe there is some link between a session cookie and `local.users`, which makes the second one to become empty when a session cookie gets deleted, but i do not see it

**Reply by Sirneij** [reply]
Hi,

No documentation states that. What I meant by "automatically" is as soon as you "eat" the cookie, all requests to the server are rejected. Since they are rejected, the user's data is unknown. You can check this out by logging the `user` in `layout.server.ts`. At least, when I wrote this, I found it `undefined` just after the user was logged out. Could you help verify it?

Thanks.
